### PR TITLE
Make use of different names of versions consistent

### PR DESCRIPTION
**Pull request by ryuyu**
Normalize use of VerbatimVersion,NormalizedVersion, and FullVersion names

@joelverhagen @xavierdecoster

### CODE REVIEW

File: tests/NuGet.IndexingTests/TestSupport/Constants.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public class Constants
         public static readonly string LucenePropertyProjectUrl = "ProjectUrl";
         public static readonly string LucenePropertySummary = "Summary";
         public static readonly string LucenePropertyTitle = "Title";
-        public static readonly string LucenePropertyVersion = "Version";
+        public static readonly string LucenePropertyVersion = "FullVersion";
```
Comment (xavierdecoster) [inline tests/NuGet.IndexingTests/TestSupport/Constants.cs]: Did you mean `LucenePropertyFullVersion`?

**Comment by joelverhagen** [comment]
Looks really good. We do need:

- unit tests in Search service and the lucene index creation code
- functional test on search service

File: src/NuGet.Indexing/Extraction/DocumentCreator.cs [hunk 2]
```diff
@@ -99,41 +99,24 @@ private static void AddId(Document document, IDictionary<string, string> package
...
-                    NuGetVersion nuGetVersion;
-                    if (NuGetVersion.TryParse(verbatimVersion, out nuGetVersion))
-                    {
-                        version = nuGetVersion.ToNormalizedString();
-                    }
-                    else
-                    {
-                        errors.Add($"Unable to parse '{MetadataConstants.VersionPropertyName}' as NuGetVersion.");
-                    }
+                    AddField(document, LuceneConstants.NormalizedVersionPropertyName, parsedVerbatimVersion.ToNormalizedString(), Field.Index.NOT_ANALYZED);
```
Comment (loic-sharma) [inline src/NuGet.Indexing/Extraction/DocumentCreator.cs]: Previously, the `NormalizedVersionPropertyName` was marked as `Field.Index.ANALYZED`, now it's `Field.Index.NOT_ANALYZED`. Is that intended? What is the difference between the two?

Comment (joelverhagen) [inline src/NuGet.Indexing/Extraction/DocumentCreator.cs]: Nice catch. This needs to be `ANALYZED` if we want the `version:1.0.0` to work (we do!)

Comment (joelverhagen) [inline src/NuGet.Indexing/Extraction/DocumentCreator.cs]: Also, we now need a functional test to verify searching by version work.

**Comment by ryuyu** [comment]
Test have been corrected. It appears we have tests that cover this already. Note that this is not a behavioural change, so the tests we have should validate the same.

**Pull request merged**